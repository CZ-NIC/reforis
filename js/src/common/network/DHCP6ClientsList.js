/*
 * Copyright (C) 2019-2021 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import PropTypes from "prop-types";
import moment from "moment";
import { formFieldsSize } from "foris";

DHCP6ClientsList.propTypes = {
    ipv6clients: PropTypes.arrayOf(PropTypes.object),
};

export default function DHCP6ClientsList({ ipv6clients }) {
    return (
        <div className={formFieldsSize}>
            <h2>{_("IPv6 DHCP Client List")}</h2>
            <p>
                {_(
                    "This list contains all devices that are connected to the network through wired or wireless connections using IPv6."
                )}
            </p>
            {ipv6clients.length === 0 ? (
                <p className="text-muted text-center">
                    {_("No clients found.")}
                </p>
            ) : (
                <div className="table-responsive">
                    <table className="table table-hover">
                        <thead className="thead-light">
                            <tr className="text-left">
                                <th>{_("Hostname")}</th>
                                <th>{_("IPv6 Address")}</th>
                                <th>{_("DUID")}</th>
                                <th className="text-center">{_("Expires")}</th>
                            </tr>
                        </thead>
                        <tbody>
                            {ipv6clients.map((client) => (
                                <DHCP6ClientsListItem
                                    key={client.ipv6}
                                    {...client}
                                />
                            ))}
                        </tbody>
                    </table>
                </div>
            )}
        </div>
    );
}

DHCP6ClientsListItem.propTypes = {
    ipv6: PropTypes.string.isRequired,
    expires: PropTypes.number.isRequired,
    duid: PropTypes.string.isRequired,
    hostname: PropTypes.string.isRequired,
};

function DHCP6ClientsListItem({ ipv6, expires, duid, hostname }) {
    return (
        <tr className="text-left">
            <td>{hostname}</td>
            <td>{ipv6}</td>
            <td>{duid}</td>
            <td className="text-center">
                {moment.unix(expires).format("YYYY-MM-DD HH:mm")}
            </td>
        </tr>
    );
}
