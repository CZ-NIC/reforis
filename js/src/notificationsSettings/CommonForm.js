/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import PropTypes from "prop-types";

import { TextInput } from "foris";
import { Select } from "foris";
import { CheckBox } from "foris";

import HELP_TEXTS from "./helpTexts";

export const SEVERITY_OPTIONS = {
    // '0': _?,
    1: _("Reboot is required"),
    2: _("Reboot or attention is required"),
    3: _("Reboot or attention is required or update was installed"),
};

CommonForm.propTypes = {
    formData: PropTypes.shape({
        to: PropTypes.string,
        severity_filter: PropTypes.oneOf(
            Object.keys(SEVERITY_OPTIONS).map((key) => parseInt(key)),
        ).isRequired,
        send_news: PropTypes.bool.isRequired,
    }).isRequired,
    formErrors: PropTypes.shape({ to: PropTypes.string }),
    setFormValue: PropTypes.func.isRequired,
};

CommonForm.defaultProps = {
    setFormValue: () => {
    },
    formData: {},
    formErrors: {},
};

export default function CommonForm({
    formData, formErrors, setFormValue, ...props
}) {
    return (
        <>
            <TextInput
                label={_("Recipient's email")}
                value={formData.to || ""}
                error={formErrors.to}
                helpText={HELP_TEXTS.common.to}
                required

                onChange={setFormValue((value) => ({ common: { to: { $set: value } } }))}

                {...props}
            />
            <Select
                label={_("Importance")}
                value={formData.severity_filter}
                choices={SEVERITY_OPTIONS}

                onChange={setFormValue(
                    (value) => ({ common: { severity_filter: { $set: value } } }),
                )}

                {...props}
            />
            <CheckBox
                label={_("Send news")}
                checked={formData.send_news}
                helpText={HELP_TEXTS.common.send_news}

                onChange={setFormValue((value) => ({ common: { send_news: { $set: value } } }))}

                {...props}
            />
        </>
    );
}
