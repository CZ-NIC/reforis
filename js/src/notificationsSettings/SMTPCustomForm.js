/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import PropTypes from "prop-types";

import { Select } from "foris";
import { TextInput } from "foris";
import { NumberInput } from "foris";
import { PasswordInput } from "foris";
import { EmailInput } from "foris";

import HELP_TEXTS from "./helpTexts";

const SECURITY_CHOICES = {
    none: _("None"),
    ssl: _("SSL"),
    starttls: _("STARTTLS"),
};

SMTPCustomForm.propTypes = {
    formData: PropTypes.shape({
        from: PropTypes.string,
        host: PropTypes.string,
        port: PropTypes.number,
        security: PropTypes.oneOf(["none", "ssl", "starttls"]),
        user: PropTypes.string,
        password: PropTypes.string,
    }).isRequired,
    formErrors: PropTypes.shape({
        from: PropTypes.string,
        host: PropTypes.string,
        port: PropTypes.string,
        user: PropTypes.string,
        password: PropTypes.string,
    }),
    setFormValue: PropTypes.func.isRequired,
};

SMTPCustomForm.defaultProps = {
    setFormValue: () => {
    },
    formData: {},
    formErrors: {},
};

export default function SMTPCustomForm({
    formData, formErrors, setFormValue, ...props
}) {
    return (
        <>
            <h4>{_("SMTP settings")}</h4>
            <EmailInput
                label={_("Sender address (From)")}
                value={formData.from || ""}
                error={formErrors.from}
                helpText={HELP_TEXTS.smtp_custom.from}
                placeholder="router@example.com"
                required

                onChange={setFormValue((value) => ({ smtp_custom: { from: { $set: value } } }))}

                {...props}
            />
            <TextInput
                label={_("Server address")}
                value={formData.host || ""}
                error={formErrors.host}
                placeholder="example.com"
                required

                onChange={setFormValue((value) => ({ smtp_custom: { host: { $set: value } } }))}

                {...props}
            />
            <NumberInput
                label={_("Server port")}
                value={formData.port}
                error={formErrors.port}
                min={1}
                max={65535}

                onChange={setFormValue((value) => ({ smtp_custom: { port: { $set: value } } }))}

                {...props}
            />
            <Select
                label={_("Security")}
                value={formData.security}
                choices={SECURITY_CHOICES}

                onChange={setFormValue((value) => ({ smtp_custom: { security: { $set: value } } }))}

                {...props}
            />
            <TextInput
                label={_("User")}
                value={formData.user || ""}

                onChange={setFormValue((value) => ({ smtp_custom: { user: { $set: value } } }))}

                {...props}
            />
            <PasswordInput
                label={_("Password")}
                withEye
                value={formData.password || ""}
                onChange={setFormValue((value) => ({ smtp_custom: { password: { $set: value } } }))}

                {...props}
            />

        </>
    );
}
