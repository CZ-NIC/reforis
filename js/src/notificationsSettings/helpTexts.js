/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

const HELP_TEXTS = {
    smtp_type: _("If you set SMTP provider to \"Turris\", the servers provided to members of the Turris project would be "
        + "used. These servers do not require any additional settings. If you want to set your own SMTP server, please "
        + "select \"Custom\" and enter required settings."),
    common: {
        to: _("Email address of recipient. Separate multiple addresses by comma."),
        send_news: _("Send emails about new features."),
    },
    smtp_turris: {
        sender_name: _("Name of the sender - will be used as a part of the sender's email address before the \"at\" sign."),
    },
    smtp_custom: {
        from: _("This is the address notifications are send from."),
    },
};

export default HELP_TEXTS;
