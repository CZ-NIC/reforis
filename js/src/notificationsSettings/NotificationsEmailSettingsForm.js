/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import PropTypes from "prop-types";

import { CheckBox, RadioSet } from "foris";

import HELP_TEXTS from "./helpTexts";
import CommonForm from "./CommonForm";
import SMTPTurrisForm from "./SMTPTurrisForm";
import SMTPCustomForm from "./SMTPCustomForm";

const SMTP_TYPE_CHOICES = [
    { label: _("Turris"), value: "turris" },
    { label: _("Custom"), value: "custom" },
];

NotificationsEmailSettingsForm.propTypes = {
    formData: PropTypes.oneOfType(
        [
            PropTypes.shape({}),
            PropTypes.shape({
                emails: PropTypes.shape({
                    enabled: PropTypes.bool.isRequired,
                    smtp_type: PropTypes.oneOf(["turris", "custom"]),
                    common: PropTypes.object,
                    smtp_turris: PropTypes.object,
                    smtp_custom: PropTypes.object,
                }),
            }),
        ],
    ).isRequired,
    formErrors: PropTypes.shape({}),
    setFormValue: PropTypes.func.isRequired,
};

NotificationsEmailSettingsForm.defaultProps = {
    setFormValue: () => {
    },
    formData: {},
    formErrors: {},
};

export default function NotificationsEmailSettingsForm({
    formData, formErrors, setFormValue, ...props
}) {
    let smtpForm = null;
    if (formData.smtp_type === "turris") {
        smtpForm = (
            <SMTPTurrisForm
                formData={formData.smtp_turris}
                formErrors={formErrors.smtp_turris}
                setFormValue={setFormValue}
                {...props}
            />
        );
    } else if (formData.smtp_type === "custom") {
        smtpForm = (
            <SMTPCustomForm
                formData={formData.smtp_custom}
                formErrors={formErrors.smtp_custom}
                setFormValue={setFormValue}
                {...props}
            />
        );
    }

    return (
        <>
            <h3>{_("Email notifications")}</h3>
            <CheckBox
                label={_("Enable email notifications")}
                checked={formData.enabled}
                onChange={setFormValue(
                    (value) => ({ enabled: { $set: value } }),
                )}

                {...props}
            />
            {formData.enabled
                ? (
                    <>
                        <RadioSet
                            label={_("SMTP provider")}
                            name="smtp_provider"
                            choices={SMTP_TYPE_CHOICES}
                            value={formData.smtp_type}
                            helpText={HELP_TEXTS.smtp_type}

                            onChange={setFormValue(
                                (value) => ({ smtp_type: { $set: value } }),
                            )}

                            {...props}
                        />
                        <CommonForm
                            formData={formData.common}
                            formErrors={formErrors.common}
                            setFormValue={setFormValue}

                            {...props}
                        />
                        {smtpForm}
                    </>
                )
                : null}
        </>
    );
}
