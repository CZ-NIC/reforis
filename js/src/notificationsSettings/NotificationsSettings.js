/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';

import ForisForm from '../formContainer/ForisForm';
import API_URLs from '../common/API';

import NotificationsEmailSettingsForm from './NotificationsEmailSettingsForm';
import validator from './validator';

export default function NotificationsSettings({ws}) {
    return <>
        <ForisForm
            ws={ws}
            forisConfig={{
                endpoint: API_URLs.notificationsSettings,
                wsModule: 'router_notifications',
                wsAction: 'update_email_settings',
            }}
            prepData={prepData}
            prepDataToSubmit={prepDataToSubmit}
            validator={validator}
        >
            <NotificationsEmailSettingsForm/>
        </ForisForm>
    </>
}

function prepData(formData) {
    formData = formData.emails;
    formData.common.to = formData.common.to.join(', ');
    return formData
}

function prepDataToSubmit(formData) {
    if (!formData.enabled) return {enabled: false};
    formData.common.to = formData.common.to.replace(/\s+/g, '').split(',');

    if (formData.smtp_type === 'turris')
        delete formData.smtp_custom;
    else if (formData.smtp_type === 'custom')
        delete formData.smtp_turris;
    return formData
}
