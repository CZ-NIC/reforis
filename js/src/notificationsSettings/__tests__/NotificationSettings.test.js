/*
 * Copyright (C) 2020-2021 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";

import {
    fireEvent,
    getByLabelText,
    getByText,
    render,
    wait,
    waitForElement,
} from "foris/testUtils/customTestRender";

import { WebSockets } from "foris";
import { mockJSONError } from "foris/testUtils/network";
import mockAxios from "jest-mock-axios";
import notificationsSettings, {
    testNotificationSeverityFixture2,
    testNotificationSeverityFixture3,
} from "./__fixtures__/notificationsSettings";

import NotificationsSettings from "../NotificationsSettings";
import {
    UNSAVED_CHANGES_MODAL_MESSAGE,
    SEVERITY_ALERT_MESSAGE,
} from "../TestNotification";

const ENABLE_CHECKBOX_LABEL = "Enable Email Notifications";

describe("<NotificationsSettings/>", () => {
    let NotificationsSettingsContainer;

    beforeEach(async () => {
        const webSockets = new WebSockets();
        const { container } = render(
            <>
                <NotificationsSettings ws={webSockets} />
                <div id="modal-container" />
            </>
        );
        mockAxios.mockResponse({ data: notificationsSettings() });
        NotificationsSettingsContainer = container;
        await wait(() => {
            expect(
                getByLabelText(container, ENABLE_CHECKBOX_LABEL)
            ).toBeTruthy();
        });
    });

    it("should handle error", async () => {
        const webSockets = new WebSockets();
        const { container } = render(<NotificationsSettings ws={webSockets} />);
        mockJSONError();
        await wait(() =>
            getByText(container, "An error occurred while fetching data.")
        );
    });

    it("Enabled, smtp_type:custom", () => {
        expect(NotificationsSettingsContainer).toMatchSnapshot();
    });

    it("Disabled", () => {
        fireEvent.click(
            getByLabelText(
                NotificationsSettingsContainer,
                ENABLE_CHECKBOX_LABEL
            )
        );
        expect(NotificationsSettingsContainer).toMatchSnapshot();
    });

    it("Enabled,smtp_type:turris", () => {
        fireEvent.click(
            getByLabelText(NotificationsSettingsContainer, "Turris")
        );
        expect(NotificationsSettingsContainer).toMatchSnapshot();
    });

    it("Post.", () => {
        fireEvent.click(getByText(NotificationsSettingsContainer, "Save"));

        expect(mockAxios.post).toBeCalled();
        const data = {
            common: {
                send_news: true,
                severity_filter: 1,
                to: ["some@example.com"],
            },
            enabled: true,
            smtp_custom: {
                from: "router@example.com",
                host: "example.com",
                password: "test_password",
                port: 465,
                security: "ssl",
                username: "root",
            },
            smtp_type: "custom",
        };
        expect(mockAxios.post).toHaveBeenCalledWith(
            "/reforis/api/notifications-settings",
            data,
            expect.anything()
        );
    });

    it("Post smtp_type:turris.", () => {
        fireEvent.click(
            getByLabelText(NotificationsSettingsContainer, "Turris")
        );
        fireEvent.click(getByText(NotificationsSettingsContainer, "Save"));

        expect(mockAxios.post).toBeCalled();
        const data = {
            common: {
                send_news: true,
                severity_filter: 1,
                to: ["some@example.com"],
            },
            enabled: true,
            smtp_type: "turris",
            smtp_turris: { sender_name: "turris" },
        };
        expect(mockAxios.post).toHaveBeenCalledWith(
            "/reforis/api/notifications-settings",
            data,
            expect.anything()
        );
    });

    it("Post disabled.", () => {
        fireEvent.click(
            getByLabelText(
                NotificationsSettingsContainer,
                ENABLE_CHECKBOX_LABEL
            )
        );
        fireEvent.click(getByText(NotificationsSettingsContainer, "Save"));

        expect(mockAxios.post).toBeCalled();
        const data = { enabled: false };
        expect(mockAxios.post).toHaveBeenCalledWith(
            "/reforis/api/notifications-settings",
            data,
            expect.anything()
        );
    });

    describe("<TestNotification/>", () => {
        let option;

        beforeEach(() => {
            option = getByLabelText(
                NotificationsSettingsContainer,
                /Importance/
            );
        });

        it("Displaying test notification importance warning (severity: 1)", () => {
            expect(
                NotificationsSettingsContainer.childNodes[3].childNodes[0]
                    .innerHTML
            ).toEqual(SEVERITY_ALERT_MESSAGE);
            expect(
                NotificationsSettingsContainer.childNodes[3]
            ).toMatchSnapshot();
        });

        it("No test notification importance warning (severity: 2)", () => {
            fireEvent.change(option, {
                target: {
                    value: "2",
                },
            });
            fireEvent.click(getByText(NotificationsSettingsContainer, "Save"));

            expect(mockAxios.post).toHaveBeenCalledWith(
                "/reforis/api/notifications-settings",
                testNotificationSeverityFixture2(),
                expect.anything()
            );
            expect(
                NotificationsSettingsContainer.childNodes[3]
            ).toMatchSnapshot();
        });

        it("No test notification importance warning (severity: 3)", () => {
            fireEvent.change(option, {
                target: {
                    value: "3",
                },
            });
            fireEvent.click(getByText(NotificationsSettingsContainer, "Save"));

            expect(mockAxios.post).toHaveBeenCalledWith(
                "/reforis/api/notifications-settings",
                testNotificationSeverityFixture3(),
                expect.anything()
            );
            expect(
                NotificationsSettingsContainer.childNodes[3]
            ).toMatchSnapshot();
        });

        it("Should toggle unsaved changes warning modal", async () => {
            fireEvent.change(option, {
                target: {
                    value: "2",
                },
            });
            fireEvent.click(
                getByText(
                    NotificationsSettingsContainer,
                    "Send test notification"
                )
            );
            await waitForElement(() =>
                getByText(
                    NotificationsSettingsContainer,
                    UNSAVED_CHANGES_MODAL_MESSAGE
                )
            );
            expect(
                NotificationsSettingsContainer.childNodes[3]
            ).toMatchSnapshot();
        });
    });
});
