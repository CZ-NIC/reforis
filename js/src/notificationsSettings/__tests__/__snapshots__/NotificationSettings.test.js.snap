// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationsSettings/> <TestNotification/> Displaying test notification importance warning (severity: 1) 1`] = `
<div
  id="test-notification"
>
  <h2>
    Test notification
  </h2>
  <div
    class="alert  alert-warning"
  >
    You will not receive the test notification to your email inbox with current importance level. Please raise the importance level at least to "Reboot or attention is required" and click "Save" button if you want to get this notification by email.
  </div>
  <div
    class="col-sm-12 col-lg-12 p-0 mb-3 text-right mb-3"
  >
    <button
      class="btn btn-primary  col-sm-12 col-md-3 col-lg-2"
      type="button"
    >
      Send test notification
    </button>
  </div>
</div>
`;

exports[`<NotificationsSettings/> <TestNotification/> No test notification importance warning (severity: 2) 1`] = `
<div
  id="test-notification"
>
  <h2>
    Test notification
  </h2>
  <div
    class="col-sm-12 col-lg-12 p-0 mb-3 text-right mb-3"
  >
    <button
      class="btn btn-primary  col-sm-12 col-md-3 col-lg-2"
      type="button"
    >
      Send test notification
    </button>
  </div>
</div>
`;

exports[`<NotificationsSettings/> <TestNotification/> No test notification importance warning (severity: 3) 1`] = `
<div
  id="test-notification"
>
  <h2>
    Test notification
  </h2>
  <div
    class="col-sm-12 col-lg-12 p-0 mb-3 text-right mb-3"
  >
    <button
      class="btn btn-primary  col-sm-12 col-md-3 col-lg-2"
      type="button"
    >
      Send test notification
    </button>
  </div>
</div>
`;

exports[`<NotificationsSettings/> <TestNotification/> Should toggle unsaved changes warning modal 1`] = `
<div
  id="test-notification"
>
  <h2>
    Test notification
  </h2>
  <div
    class="alert  alert-warning"
  >
    You will not receive the test notification to your email inbox with current importance level. Please raise the importance level at least to "Reboot or attention is required" and click "Save" button if you want to get this notification by email.
  </div>
  <div
    class="col-sm-12 col-lg-12 p-0 mb-3 text-right mb-3"
  >
    <button
      class="btn btn-primary  col-sm-12 col-md-3 col-lg-2"
      type="button"
    >
      Send test notification
    </button>
  </div>
</div>
`;

exports[`<NotificationsSettings/> Disabled 1`] = `
<div>
  <h1>
    Notifications
  </h1>
  <p>
    Your device can send you e-mail notifications about required reboots, important events, installed updates and new features according to your settings. When done with configuration, you can test it by pressing Send test notification.
  </p>
  <div
    class="card p-4 col-sm-12 col-lg-12 p-0 mb-3"
  >
    <form>
      <h2>
        Email Notifications
      </h2>
      <div
        class="form-group"
      >
        <div
          class="custom-control custom-switch custom-control-inline"
        >
          <input
            checked=""
            class="custom-control-input"
            id="1"
            type="checkbox"
          />
          <label
            class="custom-control-label"
            for="1"
          >
            Enable email notifications
          </label>
        </div>
      </div>
      <div
        class="text-right"
      >
        <button
          class="btn btn-primary  col-sm-12 col-md-3 col-lg-2"
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>
  <div
    id="test-notification"
  />
  <div
    id="modal-container"
  />
</div>
`;

exports[`<NotificationsSettings/> Enabled, smtp_type:custom 1`] = `
<div>
  <h1>
    Notifications
  </h1>
  <p>
    Your device can send you e-mail notifications about required reboots, important events, installed updates and new features according to your settings. When done with configuration, you can test it by pressing Send test notification.
  </p>
  <div
    class="card p-4 col-sm-12 col-lg-12 p-0 mb-3"
  >
    <form>
      <h2>
        Email Notifications
      </h2>
      <div
        class="form-group"
      >
        <div
          class="custom-control custom-switch custom-control-inline"
        >
          <input
            checked=""
            class="custom-control-input"
            id="1"
            type="checkbox"
          />
          <label
            class="custom-control-label"
            for="1"
          >
            Enable email notifications
          </label>
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          class="d-block"
          for="2"
        >
          SMTP provider
        </label>
        <div
          class="custom-control custom-radio custom-control-inline"
        >
          <input
            class="custom-control-input"
            id="smtp_provider-0"
            name="smtp_provider"
            type="radio"
            value="turris"
          />
          <label
            class="custom-control-label"
            for="smtp_provider-0"
          >
            Turris
          </label>
        </div>
        <div
          class="custom-control custom-radio custom-control-inline"
        >
          <input
            checked=""
            class="custom-control-input"
            id="smtp_provider-1"
            name="smtp_provider"
            type="radio"
            value="custom"
          />
          <label
            class="custom-control-label"
            for="smtp_provider-1"
          >
            Custom
          </label>
        </div>
        <small
          class="form-text text-muted"
        >
          If you set SMTP provider to "Turris", the servers provided to members of the Turris project would be used. These servers do not require any additional settings. If you want to set your own SMTP server, please select "Custom" and enter required settings.
        </small>
      </div>
      <div
        class="form-group"
      >
        <label
          for="3"
        >
          Recipient's email
        </label>
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="3"
            required=""
            type="text"
            value="some@example.com"
          />
        </div>
        <small
          class="form-text text-muted"
        >
          Email address of recipient. Separate multiple addresses by comma.
        </small>
      </div>
      <div
        class="form-group"
      >
        <label
          for="4"
        >
          Importance
        </label>
        <select
          class="custom-select"
          id="4"
        >
          <option
            value="1"
          >
            Reboot is required
          </option>
          <option
            value="2"
          >
            Reboot or attention is required
          </option>
          <option
            value="3"
          >
            Reboot or attention is required or update was installed
          </option>
        </select>
      </div>
      <div
        class="form-group"
      >
        <div
          class="custom-control custom-checkbox "
        >
          <input
            checked=""
            class="custom-control-input"
            id="5"
            type="checkbox"
          />
          <label
            class="custom-control-label"
            for="5"
          >
            Send news
            <small
              class="form-text text-muted"
            >
              Send emails about new features.
            </small>
          </label>
        </div>
      </div>
      <h3>
        SMTP Settings
      </h3>
      <div
        class="form-group"
      >
        <label
          for="6"
        >
          Sender address (From)
        </label>
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="6"
            placeholder="router@example.com"
            required=""
            type="email"
            value="router@example.com"
          />
        </div>
        <small
          class="form-text text-muted"
        >
          This is the address notifications are send from.
        </small>
      </div>
      <div
        class="form-group"
      >
        <label
          for="7"
        >
          Server address
        </label>
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="7"
            placeholder="example.com"
            required=""
            type="text"
            value="example.com"
          />
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          for="8"
        >
          Server port
        </label>
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="8"
            max="65535"
            min="1"
            type="number"
            value="465"
          />
          <div
            class="input-group-append"
          >
            <button
              aria-label="Increase"
              class="btn btn-outline-secondary"
              type="button"
            >
              <i
                class="fas fa-plus"
              />
            </button>
            <button
              aria-label="Decrease"
              class="btn btn-outline-secondary"
              type="button"
            >
              <i
                class="fas fa-minus"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          for="9"
        >
          Security
        </label>
        <select
          class="custom-select"
          id="9"
        >
          <option
            value="none"
          >
            None
          </option>
          <option
            value="ssl"
          >
            SSL
          </option>
          <option
            value="starttls"
          >
            STARTTLS
          </option>
        </select>
      </div>
      <div
        class="form-group"
      >
        <label
          for="10"
        >
          Username
        </label>
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="10"
            type="text"
            value="root"
          />
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          for="11"
        >
          Password
        </label>
        <div
          class="input-group"
        >
          <input
            autocomplete="new-password"
            class="form-control"
            id="11"
            type="password"
            value="test_password"
          />
          <div
            class="input-group-append"
          >
            <button
              class="input-group-text"
              type="button"
            >
              <i
                class="fa fa-eye"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="text-right"
      >
        <button
          class="btn btn-primary  col-sm-12 col-md-3 col-lg-2"
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>
  <div
    id="test-notification"
  >
    <h2>
      Test notification
    </h2>
    <div
      class="alert  alert-warning"
    >
      You will not receive the test notification to your email inbox with current importance level. Please raise the importance level at least to "Reboot or attention is required" and click "Save" button if you want to get this notification by email.
    </div>
    <div
      class="col-sm-12 col-lg-12 p-0 mb-3 text-right mb-3"
    >
      <button
        class="btn btn-primary  col-sm-12 col-md-3 col-lg-2"
        type="button"
      >
        Send test notification
      </button>
    </div>
  </div>
  <div
    id="modal-container"
  >
    <div
      class="modal fade"
      role="dialog"
    >
      <div
        class="modal-dialog  modal-dialog-centered"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h5
              class="modal-title"
            >
              Warning!
            </h5>
            <button
              class="close"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </button>
          </div>
          <div
            class="modal-body"
          >
            <p>
              There are some unsaved changes in the notifications settings. Do you want to discard them and test the notifications with the old settings?
            </p>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-primary "
              type="button"
            >
              Cancel
            </button>
            <button
              class="btn btn-danger"
              type="button"
            >
              Confirm
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<NotificationsSettings/> Enabled,smtp_type:turris 1`] = `
<div>
  <h1>
    Notifications
  </h1>
  <p>
    Your device can send you e-mail notifications about required reboots, important events, installed updates and new features according to your settings. When done with configuration, you can test it by pressing Send test notification.
  </p>
  <div
    class="card p-4 col-sm-12 col-lg-12 p-0 mb-3"
  >
    <form>
      <h2>
        Email Notifications
      </h2>
      <div
        class="form-group"
      >
        <div
          class="custom-control custom-switch custom-control-inline"
        >
          <input
            checked=""
            class="custom-control-input"
            id="1"
            type="checkbox"
          />
          <label
            class="custom-control-label"
            for="1"
          >
            Enable email notifications
          </label>
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          class="d-block"
          for="2"
        >
          SMTP provider
        </label>
        <div
          class="custom-control custom-radio custom-control-inline"
        >
          <input
            class="custom-control-input"
            id="smtp_provider-0"
            name="smtp_provider"
            type="radio"
            value="turris"
          />
          <label
            class="custom-control-label"
            for="smtp_provider-0"
          >
            Turris
          </label>
        </div>
        <div
          class="custom-control custom-radio custom-control-inline"
        >
          <input
            checked=""
            class="custom-control-input"
            id="smtp_provider-1"
            name="smtp_provider"
            type="radio"
            value="custom"
          />
          <label
            class="custom-control-label"
            for="smtp_provider-1"
          >
            Custom
          </label>
        </div>
        <small
          class="form-text text-muted"
        >
          If you set SMTP provider to "Turris", the servers provided to members of the Turris project would be used. These servers do not require any additional settings. If you want to set your own SMTP server, please select "Custom" and enter required settings.
        </small>
      </div>
      <div
        class="form-group"
      >
        <label
          for="3"
        >
          Recipient's email
        </label>
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="3"
            required=""
            type="text"
            value="some@example.com"
          />
        </div>
        <small
          class="form-text text-muted"
        >
          Email address of recipient. Separate multiple addresses by comma.
        </small>
      </div>
      <div
        class="form-group"
      >
        <label
          for="4"
        >
          Importance
        </label>
        <select
          class="custom-select"
          id="4"
        >
          <option
            value="1"
          >
            Reboot is required
          </option>
          <option
            value="2"
          >
            Reboot or attention is required
          </option>
          <option
            value="3"
          >
            Reboot or attention is required or update was installed
          </option>
        </select>
      </div>
      <div
        class="form-group"
      >
        <div
          class="custom-control custom-checkbox "
        >
          <input
            checked=""
            class="custom-control-input"
            id="5"
            type="checkbox"
          />
          <label
            class="custom-control-label"
            for="5"
          >
            Send news
            <small
              class="form-text text-muted"
            >
              Send emails about new features.
            </small>
          </label>
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          for="17"
        >
          Sender's name
        </label>
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="17"
            type="text"
            value="turris"
          />
        </div>
        <small
          class="form-text text-muted"
        >
          Name of the sender - will be used as a part of the sender's email address before the "at" sign.
        </small>
      </div>
      <div
        class="text-right"
      >
        <button
          class="btn btn-primary  col-sm-12 col-md-3 col-lg-2"
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>
  <div
    id="test-notification"
  >
    <h2>
      Test notification
    </h2>
    <div
      class="alert  alert-warning"
    >
      You will not receive the test notification to your email inbox with current importance level. Please raise the importance level at least to "Reboot or attention is required" and click "Save" button if you want to get this notification by email.
    </div>
    <div
      class="col-sm-12 col-lg-12 p-0 mb-3 text-right mb-3"
    >
      <button
        class="btn btn-primary  col-sm-12 col-md-3 col-lg-2"
        type="button"
      >
        Send test notification
      </button>
    </div>
  </div>
  <div
    id="modal-container"
  >
    <div
      class="modal fade"
      role="dialog"
    >
      <div
        class="modal-dialog  modal-dialog-centered"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h5
              class="modal-title"
            >
              Warning!
            </h5>
            <button
              class="close"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </button>
          </div>
          <div
            class="modal-body"
          >
            <p>
              There are some unsaved changes in the notifications settings. Do you want to discard them and test the notifications with the old settings?
            </p>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-primary "
              type="button"
            >
              Cancel
            </button>
            <button
              class="btn btn-danger"
              type="button"
            >
              Confirm
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
